lib/src/wiggle_abi/req_impl.rs

```diff
@@ -584,43 +584,43 @@ impl FastlyHttpReq for Session {
 
         // asynchronously send the request
         let pending_req =
-            PendingRequest::spawn(upstream::send_request(req, backend, self.tls_config()));
+            PeekableTask::spawn(upstream::send_request(req, backend, self.tls_config()));
 
         // return a handle to the pending request
-        Ok(self.insert_pending_request(pending_req))
+        Ok(self.insert_pending_request(pending_req.await))
```

Comment:
Another `.await` would look more natural placed directly after the async expression